src/pinnable_mapped_file.cpp

```diff
@@ -230,25 +314,27 @@ void pinnable_mapped_file::setup_non_file_mapping() {
 }
 
 void pinnable_mapped_file::load_database_file(boost::asio::io_service& sig_ios) {
-   std::cerr << "CHAINBASE: Preloading \"" << _database_name << "\" database file, this could take a moment..." << std::endl;
-   char* const src = (char*)_file_mapped_region.get_address();
+   std::cerr << "CHAINBASE: Preloading \"" << _database_name << "\" database file, this could take a moment..." << '\n';
    char* const dst = (char*)_non_file_mapped_mapping;
    size_t offset = 0;
    time_t t = time(nullptr);
-   while(offset != _file_mapped_region.get_size()) {
-      memcpy(dst+offset, src+offset, _db_size_multiple_requirement);
-      offset += _db_size_multiple_requirement;
+   while(offset != _database_size) {
+      size_t copy_size = std::min(_db_size_copy_increment, _database_size - offset);
+      bip::mapped_region src_rgn(_file_mapping, bip::read_only, offset, copy_size);
+      memcpy(dst+offset, src_rgn.get_address(), copy_size);
+      offset += copy_size;
 
       if(time(nullptr) != t) {
          t = time(nullptr);
-         std::cerr << "CHAINBASE: Preloading \"" << _database_name << "\" database file, " << offset/(_file_mapped_region.get_size()/100) << "% complete..." << std::endl;
+         std::cerr << "CHAINBASE: Preloading \"" << _database_name << "\" database file, " <<
+            offset/(_file_mapped_region.get_size()/100) << "% complete..." << '\n';
```

Comment:
I'm getting a divide by zero on this line when running nodeos like
```
nodeos -e -peosio --delete-all --snapshot /home/spoon/RAMsnaps/snapshot-1287571bdd7b7337bcaac970f9cbc03b69fcd039a0784068b97152f130b483c5.bin --chain-state-db-size-mb 24576 --database-map-mode heap
```

Reply:
Fixed.